filament/backend/test/test_ComputeBasic.cpp

```diff
@@ -92,11 +96,25 @@ void main() {
 }
 )"};
 
-    std::string_view shader_msl = {R"(
-// TODO: msl test
+    std::string shader_msl = {R"(
+#include <simd/simd.h>
+#include <metal_stdlib>
+using namespace metal;
+struct Output_data {
+    float elements[1];
+};
+struct Input_data {
+    float elements[1];
+};
+constant uint3 gl_WorkGroupSize [[maybe_unused]] = uint3(16u, 1u, 1u);
+kernel void main0(device Output_data& output_data [[buffer(0)]],
+        device Input_data& input_data [[buffer(1)]],
+        uint3 gl_GlobalInvocationID [[thread_position_in_grid]]) {
+    output_data.elements[gl_GlobalInvocationID.x] = input_data.elements[gl_GlobalInvocationID.x];
```

Comment:
Since these are backend tests, I don't think you should use the `gl_` nomenclature, it's confusing. You can use any name you want here.

Reply:
Yeah that's true. This was exactly the output from Spir-V cross, but I agree it doesn't matter in this case.